seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/JsonPathTransformFactoryTest.java

```diff
@@ -16,15 +16,217 @@
 import org.junit.jupiter.api.Test;
 
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
 public class JsonPathTransformFactoryTest {
+
+    @Test
+    public void testJsonPath() {
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put(
+                JsonPathTransformConfig.COLUMNS.key(),
+                Arrays.asList(
+                        ImmutableMap.of(
+                                JsonPathTransformConfig.SRC_FIELD.key(), "data",
+                                JsonPathTransformConfig.PATH.key(), "$.f1",
+                                JsonPathTransformConfig.DEST_FIELD.key(), "f1")));
+        ReadonlyConfig config = ReadonlyConfig.fromMap(configMap);
+        CatalogTable table =
+                CatalogTableUtil.getCatalogTable(
+                        "test",
+                        new SeaTunnelRowType(
+                                new String[] {"data"},
+                                new SeaTunnelDataType[] {BasicType.STRING_TYPE}));
+        JsonPathTransform transform =
+                new JsonPathTransform(JsonPathTransformConfig.of(config), table);
+
+        CatalogTable outputTable = transform.getProducedCatalogTable();
+        SeaTunnelRow outputRow = transform.map(new SeaTunnelRow(new Object[] {"{\"f1\": 1}"}));
+        Assertions.assertEquals(
+                "1", outputRow.getField(outputTable.getSeaTunnelRowType().indexOf("f1")));
+    }
+
     @Test
-    public void testOptionRule() {
-        JsonPathTransformFactory jsonPathTransformFactory = new JsonPathTransformFactory();
-        Assertions.assertNotNull(jsonPathTransformFactory.optionRule());
+    public void testErrorHandleWay() {
+        Map<String, Object> configMap = new HashMap<>();
+        configMap.put(
+                JsonPathTransformConfig.COLUMNS.key(),
+                Arrays.asList(
+                        ImmutableMap.of(
+                                JsonPathTransformConfig.SRC_FIELD.key(), "data",
+                                JsonPathTransformConfig.PATH.key(), "$.f1",
+                                JsonPathTransformConfig.DEST_FIELD.key(), "f1")));
+        ReadonlyConfig config = ReadonlyConfig.fromMap(configMap);
+        CatalogTable table =
+                CatalogTableUtil.getCatalogTable(
+                        "test",
+                        new SeaTunnelRowType(
+                                new String[] {"data"},
+                                new SeaTunnelDataType[] {BasicType.STRING_TYPE}));
+        JsonPathTransform transform =
+                new JsonPathTransform(JsonPathTransformConfig.of(config), table);
+
+        CatalogTable outputTable = transform.getProducedCatalogTable();
+        SeaTunnelRow outputRow;
+        try {
+            outputRow = transform.map(new SeaTunnelRow(new Object[] {"{\"f2\": 1}"}));
+            Assertions.fail("should throw exception");
+        } catch (Exception e) {
+            // ignore
+        }
```

Comment:
The `Assertions.fail("should throw exception");` will be catch too?
I think you can use `Assertions.assertThrows()`

Reply:
Assertions.fail throw Error will not be Exception caught

Reply:
fixed